chunky/src/java/se/llbit/chunky/resources/OctreeFileFormat.java

```diff
@@ -32,9 +32,9 @@
    * Load octrees and grass/foliage textures from a file.
    *
    * @param in input stream for the file to load the scene from.
-   * @param forceNodeBased Forces the octrees to be loaded as NodeBasedOctree
+   * @param impl The octree implementation to use
    */
-  public static OctreeData load(DataInputStream in, boolean forceNodeBased) throws IOException {
+  public static OctreeData load(DataInputStream in, Octree.ImplementationEnum impl) throws IOException {
```

Comment:
Now that I think about it… instead of having an enum with possible octree implementations and passing that around, what about an `OctreeFactory` that can be configured to return different octree implementations or switches altogether (e.g. by a plugin)?

Might be too over-engineered though.

Reply:
Changed, a string is passed around which refers to a factory that must be registered before (possibly by a plug-in)